app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt

```diff
@@ -86,7 +88,12 @@ class TabDataRepository @Inject constructor(
     }
 
     override suspend fun selectByUrlOrNewTab(url: String) {
-        val tabId = tabsDao.selectTabByUrl(url)
+        var searchUrl = url
+        if (useOurAppDetector.isUseOurAppUrl(url)) {
+            searchUrl = "%${UseOurAppDetector.USE_OUR_APP_DOMAIN}%"
```

Comment:
If the URL is from the use our app domain we try to match by the domain rather than the full URL now. This code only gets executed when a shortcut is used so the chances to have a false positive are quite slim.